tensordict/prototype/tensorclass.py

```diff
@@ -302,16 +343,32 @@ def _setitem(self, item, value):
         raise ValueError("Invalid indexing arguments.")
     if not isinstance(value, self.__class__):
         raise ValueError("__setitem__ is only allowed for same-class assignement")
+    for key, val in self.non_tensordict.items():
```

Comment:
what are you trying to do here?

Reply:
We check if all the non_tensor values are same before setting the item

Reply:
ok, please add a comment.